projects/plugins/jetpack/extensions/blocks/contact-form/util/form.js

```diff
@@ -0,0 +1,37 @@
+import { store as blockEditorStore } from '@wordpress/block-editor';
+import { createBlock } from '@wordpress/blocks';
+import { useDispatch, useSelect } from '@wordpress/data';
+import { useEffect } from '@wordpress/element';
+import { __ } from '@wordpress/i18n';
+
+export const validateFormWrapper = ( { attributes, clientId, name } ) => {
+	const FORM_BLOCK_NAME = 'jetpack/contact-form';
+	const BUTTON_BLOCK_NAME = 'jetpack/button';
+	const SUBMIT_BUTTON_ATTR = {
+		text: __( 'Submit', 'jetpack' ),
+		element: 'button',
+		lock: { remove: true },
+	};
+
+	// eslint-disable-next-line react-hooks/rules-of-hooks
+	const { replaceBlock } = useDispatch( blockEditorStore );
+
+	// eslint-disable-next-line react-hooks/rules-of-hooks
+	const parents = useSelect( select => {
+		return select( blockEditorStore ).getBlockParentsByBlockName( clientId, FORM_BLOCK_NAME );
+	} );
+
+	// eslint-disable-next-line react-hooks/rules-of-hooks
```

Comment:
Maybe this extraction would be the opportunity to turn this into a proper hook, and thus getting rid of the warnings?

Reply:
Is [f8e8dd7](https://github.com/Automattic/jetpack/pull/28010/commits/f8e8dd7b77464565a87adf0c0271cf709acafb73) what you're suggesting?

Reply:
Yes! This and the follow-up commit make things a lot cleaner in my opinion, wouldn't you agree?

Reply:
Yes, I agree! My question was more of a confidence check 🙂